src/main/java/org/terasology/ligthandshadow/logic/CardSystem.java

```diff
@@ -89,31 +64,30 @@ public void placeCard(ActivateEvent event, EntityRef entity) {
             return;
         }
 
-        Vector3f offset = new Vector3f(event.getHitPosition());
-        offset.sub(new Vector3f(targetBlockComponent.getPosition(new Vector3i())));
-        Side offsetDir = Side.inDirection(offset);
-
         Vector3i primePos = new Vector3i(targetBlockComponent.getPosition(new Vector3i()));
-        primePos.add(offsetDir.direction());
 
         Block primeBlock = worldProvider.getBlock(primePos);
-        if (!primeBlock.isReplacementAllowed()) {
+        boolean exit = true;
+        for (String s : primeBlock.getBlockFamily().getCategories()) {
+            if (s.equals("soil")) {
+                exit = false;
+                break;
+            }
+        }
+        if (exit) {
             event.consume();
             return;
         }
```

Comment:
Why this check for soil?

Reply:
This is so that the card is only placed above the ground (soil) rather than on top of grass, flowers, the bases etc.

Reply:
in that case, I'd rather go for an attribute such as "solid"... I think you could use the block property check `isPenetrable` here 🤔 (see https://github.com/MovingBlocks/Terasology/blob/develop/engine/src/main/java/org/terasology/engine/world/block/Block.java#L230)

Reply:
I don't think that check would be sufficient for more specific scenario's such as when a player tries to place the card on the base and the floating platform. Not sure but I will test it out later.

Reply:
do we want to forbid to place anything on the base? consider that you would also need to check something like that for everything that is placeable then.
on the floating platform, a player without cheat permissions should not be able to get a card in the first place 🤔

Reply:
the shop is available even on the floating platform. By placing the cards on the base it becomes possible to barricade the flag itself which might make it too difficult to get the flag (while breaking down the barrier the player's defenses become too low).

Reply:
> the shop is available even on the floating platform

that's something I'm not entirely convinced we want (I mean that's what we have the pre-game phase for, right?) to be honest...

> it becomes possible to barricade the flag itself

I'd say this is exactly the kind of defense strategy that the cards could be used for and I don't see too much harm in that in general.
In the end it's a question of balancing (which is not reasonable to do at this point in time - yet). If in comparison to other mechanisms the cards become too powerful, we can still tweak that.
Restricting the placement is one screw you can turn, but there's also others, i.e. card price, max number of cards you can buy, etc ... But IMO restricting that already now would be premature optimization.